common/djangoapps/track/middleware.py

```diff
@@ -204,3 +205,38 @@ def process_response(self, _request, response):
             pass
 
         return response
+
+    def truncate_openended_responses(self, event):
+        """
+        Truncates the tracking log entry for problems for which the
+        student response can be long winded.
+
+        event: json, Constructed from POST and GET request params
+        """
+        event_json = json.loads(event)
+        event_post_keys = event_json['POST'].keys()
+
+        if 'student_answer' in event_json['POST'] and isinstance(event_json['POST']['student_answer'], list):
+            student_answer = event_json['POST']['student_answer']
+            truncated_answers = []
+            for answer in student_answer:
+                chunk_length = TRUNCATION_LENGTH / len(student_answer)
+                truncated_answer = answer[:chunk_length]
+                truncated_answers.append(truncated_answer)
+            event_json['POST']['student_answer'] = truncated_answers
+            return json.dumps(event_json)
+        for key in event_post_keys:
+            if key[:15] == '{"submission":[':
+                event_json['POST'].pop(key)
+                submission = json.loads(key)
+                truncated_prompts = []
+                for prompt_submission in submission['submission']:
+                    truncated_prompt = prompt_submission[:int(TRUNCATION_LENGTH / len(submission['submission']))]
```

Comment:
Can we split this out into multiple statements like above?

```python
chunk_length = TRUNCATION_LENGTH / len(submission['submission'])
truncated_prompt = prompt_submission[:chunk_length]
```

Nested statements like this are a bear to step through while debugging.